mono/metadata/appdomain.c

```diff
@@ -210,12 +209,16 @@ create_domain_objects (MonoDomain *domain)
 	mono_field_static_set_value (string_vt, string_empty_fld, empty_str);
 	domain->empty_string = empty_str;
 
-	bitconverter_isle_fld = mono_class_get_field_from_name_full (mono_defaults.bitconverter_class, "IsLittleEndian", NULL);
-	if (bitconverter_isle_fld) {
-		gint x = 1;
-		MonoBoolean is_le = *((char*) &x); // 00 00 00 01 for little-endian and 01 00 00 00 for big-endian
-		MonoVTable *bitconverter_vt = mono_class_vtable_checked (domain, mono_defaults.bitconverter_class, error);
-		mono_field_static_set_value (bitconverter_vt, bitconverter_isle_fld, &is_le);
+	/* Initialize BitConverter.IsLittleEndian */
+	MonoClass *bitconverter_class = mono_class_try_load_from_name (mono_defaults.corlib, "System", "BitConverter")
+	if (bitconverter_class) {
+		MonoClassField *bitconverter_isle_fld = mono_class_get_field_from_name_full (bitconverter_class, "IsLittleEndian", NULL);
+		if (bitconverter_isle_fld) {
+			gint x = 1;
+			MonoBoolean is_le = G_BYTE_ORDER == G_BIG_ENDIAN ? FALSE : TRUE;
```

Comment:
`const static MonoBoolean is_le = (G_BYTE_ORDER != G_BIG_ENDIAN);`
And fix `mono_copy_value` and `mono_field_static_set_value` to take `const void*`  or  `gconstpointer`.